### PR TITLE
Infra: production pipeline (step 1)

### PR DESCRIPTION
**Pull request by guifry**
This PR creates the production environment and deployment capabilities in Pulumi.

- DB deletion protection in production
- Environment follows same naming convention as in staging: ${env}-consent-api
- Fixes a bug in the resource name formatting
- Sanitise the branch name for when there are slashes
- Create a "production" stack to deploy the production environment

### CODE REVIEW

File: infra/deploy_service.py [hunk 1]
```diff
@@ -284,9 +284,14 @@ def main():
...
 
+    if args.env == "production":
+        # Production doesn't support branch environments
+        args.branch = None
+
     stack_name = args.env
-    sanitised_branch = args.branch.replace("/", "-") if args.branch else ""
+    sanitised_branch = None
     if args.branch:
+        sanitised_branch = args.branch.replace("/", "-") if args.branch else ""
```
Comment (nacnudus) [inline infra/deploy_service.py]: Isn't `if args.branch` redundant?

**Review by nacnudus — CHANGES_REQUESTED** [review]
This seems to differ somewhat from the PR description.

- [x] DB deletion protection in production
- [ ] ~Follows the same pattern as naming convention as in staging: ${env}-consent-api~ Does this PR include this work?
- [ ] ~Format the resource names to take into account the max length imposed by GCP~ I think this was already implemented, and this PR includes a bugfux for when `trimmable=None`
- [x] Sanitise the branch name for when there are slashes
- [x] Create a "production" stack to deploy the production environment

One small comment on the code itself.

File: infra/deploy_service.py (hunk 1 above)
Comment (guifry) [reply infra/deploy_service.py]: Good shout, amended

**Review by andyhd — APPROVED** [review]
Looks good 👍

**Comment by guifry** [comment]
> This seems to differ somewhat from the PR description.
> 
> * [x]  DB deletion protection in production
> * [ ]  ~Follows the same pattern as naming convention as in staging: ${env}-consent-api~ Does this PR include this work?
> * [ ]  ~Format the resource names to take into account the max length imposed by GCP~ I think this was already implemented, and this PR includes a bugfux for when `trimmable=None`
> * [x]  Sanitise the branch name for when there are slashes
> * [x]  Create a "production" stack to deploy the production environment
> 
> One small comment on the code itself.

Cheers, indeed the resource name formatting is a bugfix.
Worth mentioning the naming convention as we are moving away from the previous prod environment that had been manually set up

image

**Review by nacnudus — APPROVED** [review]

**Pull request merged**